src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/SignatureBuilder.cs

```diff
@@ -400,6 +400,7 @@ private void EmitArrayTypeSignature(ArrayType type, SignatureContext context)
             bool isInstantiatingStub)
         {
             uint flags = 0;
+            bool useConstraint = (method.ConstrainedType != null && method.ConstrainedType != method.OwningType);
```

Comment:
I don't understand why you are dropping the constraint at all. It is necessary to get the runtime to resolve the dispatch in some circumstances. If it causes the runtime to crash, that's just a reason to figure out why that's failing, and fix it. Most likely, we will need to provide logic in the runtime which will call ResolveVirtualStaticMethod from within the fixup resolver. Alternatively, if the resolved method target is within the version bubble, we could create new tokens in the manner of devirtualization, but that is only really an optimization that is sometimes applicable.

Reply:
Removed in 2nd commit.